### PR TITLE
[P1 Bug][UI/UX] Address shop cursor target feedbacks

### PR DESCRIPTION
**Pull request by torranx**
## What are the changes the user will see?
- entering daily run's rewards phase should no longer soft lock when the shop cursor target is set to `Shop`
- changed "Items" label to "Rewards"
- set the default option to `Rewards`
- set the default option to be left-most/first option

## Why am I making these changes?
Addressing feedbacks:
- P1: Game can softlock if it's set to "Shop" before buyable shop items introduced - in Daily Run mode
- Some of the terms are ambiguous. "Items" can refer to the buyable items, the free ones, or transferring items. Someone suggested "Rewards"
- Move the default option back to the original please (Rewards)
- The default most common option (Rewards) should be the first option

## What are the changes from a developer perspective?
- added an index mapping of options-enum to avoid the risk of mismatching the option keys and the shop cursor targets enum
- added condition to check the current game mode and the current set shop cursor target option

### Screenshots/Videos
Updated options
Screenshot 2024-09-04 at 4 02 25 AM

Daily run no longer softlocks when shop cursor target is set to `Shop`

https://github.com/user-attachments/assets/30f61a8c-c947-41a0-b753-5f35bbd4e97c

## How to test the changes?
set settings, enter rewards/shop phase

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by f-fsantos — DISMISSED** [review]

**Review by f-fsantos — APPROVED** [review]

File: src/test/items/dire_hit.test.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { Button } from "#app/enums/buttons";
 import { CommandPhase } from "#app/phases/command-phase";
 import { NewBattlePhase } from "#app/phases/new-battle-phase";
 import { TurnInitPhase } from "#app/phases/turn-init-phase";
+import { ShopCursorTarget } from "#app/enums/shop-cursor-target.js";
```
Comment (DayKev) [inline src/test/items/dire_hit.test.ts:16]:
```suggestion
import { ShopCursorTarget } from "#app/enums/shop-cursor-target";
```

File: src/test/items/temp_stat_stage_booster.test.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import ModifierSelectUiHandler from "#app/ui/modifier-select-ui-handler";
 import { TurnInitPhase } from "#app/phases/turn-init-phase";
 import { BattleEndPhase } from "#app/phases/battle-end-phase";
 import { EnemyCommandPhase } from "#app/phases/enemy-command-phase";
+import { ShopCursorTarget } from "#app/enums/shop-cursor-target.js";
```
Comment (DayKev) [inline src/test/items/temp_stat_stage_booster.test.ts:19]:
```suggestion
import { ShopCursorTarget } from "#app/enums/shop-cursor-target";
```

**Comment by frutescens** [comment]
What happens to everyone's settings before the PR?

**Comment by torranx** [comment]
> What happens to everyone's settings before the PR?

You mean after? The game won't break but I think it will reset back to the default option because I changed the key REROLL_TARGET to SHOP_CURSOR_TARGET

File: src/locales/ja/settings.json [hunk 3]
```diff
@@ -100,7 +100,7 @@
   "moveTouchControls": "タッチ移動操作",
   "shopOverlayOpacity": "ショップオーバレイ不透明度",
   "shopCursorTarget": "ショップカーソル初位置",
-  "items": "アイテム",
+  "rewards": "アイテム",
```
Comment (Chapybara-jp) [inline src/locales/ja/settings.json:103]:
```suggestion
  "rewards": "ご褒美",
```

**Review by Chapybara-jp — DISMISSED** [review]

File: src/locales/de/settings.json [hunk 4]
```diff
@@ -100,7 +100,7 @@
   "moveTouchControls": "Bewegung Touch Steuerung",
   "shopOverlayOpacity": "Shop Overlay Deckkraft",
   "shopCursorTarget": "Shop-Cursor Ziel",
-  "items": "Items",
+  "rewards": "Items",
```
Comment (CodeTappert) [inline src/locales/de/settings.json:103]:
```suggestion
  "rewards": "Belohnungen",
```


Please check if thats to long

**Comment by torranx** [comment]
> In my local instance, I set the cursor to Reroll from Check Team and then checked out this PR. It then changed my settings to Shop. 

That is an expected behavior with what we have now since the setting's key changed.

> I do think there should be a way to transition current settings to old b/c otherwise we'll run into similar issues from before.

What similar issues from before exactly, only "issue" here is that the users will only have to re-set the option they want? I agree there should be a way for us to transition a key to another.

File: src/locales/de/settings.json (hunk 4 above)
Comment (torranx) [reply src/locales/de/settings.json:103]: it is
Screenshot 2024-09-04 at 5 54 23 AM

File: src/system/settings/settings.ts [hunk 5]
```diff
@@ -53,6 +55,40 @@ const AUTO_DISABLED: SettingOption[] = [
...
+    label: i18next.t("settings:shop")
+  },
+  {
+    value: "Reroll",
+    label: i18next.t("settings:reroll")
+  },
+  {
+    value: "Check Team",
+    label: i18next.t("settings:checkTeam")
+  }
```
Comment (Tempo-anon) [inline src/system/settings/settings.ts:74]: minor nit: this and the switch below have all the values in different orders from the way the enum is defined. Ideally they should all be in the same order as they are displayed in game

**Review by Tempo-anon — APPROVED** [review]

File: src/system/settings/settings.ts (hunk 5 above)
Comment (torranx) [reply src/system/settings/settings.ts:74]: that is what the index mapping I added is for, the enums matches the rows in the modifier select phase ie `0` being the row where Reroll is - so the only thing I can change here is to match the switch with the value of `options`

**Review by Tempo-anon — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Review by f-fsantos — APPROVED** [review]

**Pull request merged**